Application/src/main/java/control/DettagliEventoServlet.java

```diff
@@ -1,50 +1,128 @@
-import java.sql.Connection;
-import java.sql.ResultSet;
-import java.util.Collection;
-import java.sql.PreparedStatement;
 
+import java.io.IOException;
+import java.util.*;
+import javax.servlet.RequestDispatcher;
 import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
-import model.ConDB;
-import model.evento.Evento;
-import model.evento.EventoService;
+import javax.servlet.http.HttpSession;
+import model.evento.*;
 import model.utente.*;
+import model.Valutazione.*;
+import model.prenotazione.*;
 
 @WebServlet("/DettagliEvento")
 public class DettagliEventoServlet extends HttpServlet {
     private static final long serialVersionUID = 1L;
-    
+
     private EventoService eventoService = new EventoService(); 
     private PrenotazioneService prenotazioneService = new PrenotazioneService();
+    private ValutazioneService valutazioneService = new ValutazioneService();
+    private UtenteService utenteService = new UtenteService();
 
     public DettagliEventoServlet() {
         super();
     }
 
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        String idEvento = request.getParameter("id");	//ID Evento passato nell'url
-        
-        final String redirectedPage = "dettagliEvento.jsp";
-        
-        Evento e = eventoService.dettagli_evento(idEvento);
-        Collection<UtenteBean> utenti = prenotazioneService.calcola_partecipanti(Integer.valueOf(idEvento));
-        
-        if (e == null) {
-        	response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Errore interno del server (Dettagli Evento).");
-        	return;
+        try {
+            // Recupera i parametri dalla richiesta
+            String idEventoParam = request.getParameter("id");
+            String putValutazioneParam = request.getParameter("putValutazione");
+
+            // Validazione dei parametri
+            if (idEventoParam == null || idEventoParam.isEmpty()) {
+                throw new IllegalArgumentException("ID evento non fornito o non valido.");
+            }
+
+            int idEvento;
+            try {
+                idEvento = Integer.parseInt(idEventoParam);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("ID evento non valido.");
+            }
+
+            if (idEvento <= 0) {
+                throw new IllegalArgumentException("ID evento deve essere maggiore di 0.");
+            }
+
+            // Controlla l'esistenza dell'evento
+            if (!eventoService.esiste_evento(idEvento)) {
+                throw new IllegalArgumentException("Evento non esistente.");
+            }
+
+            // Recupera i dettagli dell'evento
+            Evento e = eventoService.dettagli_evento(idEventoParam);
```

Comment:
Alla fine cambiamo la mia funzione, quindi qua si può passare l'intero invece della stringa

```suggestion
            Evento e = eventoService.dettagli_evento(Integer.valueOf(idEventoParam));
```